libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -271,11 +285,16 @@ static CK_RV prvCheckValidSessionAndModule( const P11Session_t * pxSession )
  */
 static P11Session_t * prvSessionPointerFromHandle( const CK_SESSION_HANDLE xSession )
 {
-    /* PKCS #11 uses opaque integer handles to manage sessions.
-     * It is necessary to cast this to a pointer in order to
-     * retrieve the session. */
-    /* coverity[misra_c_2012_rule_11_4_violation] */
-    return ( P11Session_t * ) xSession;
+    P11Session_t * pxSession = NULL;
+    CK_SESSION_HANDLE xSessionHandle = xSession - 1;
```

Comment:
What if xSession is 0?

Reply:
The underlying type is an unsigned long int, so it will overflow and be too big of a number to pass the if( xSessionHandle < pkcs11configMAX_SESSIONS ) statment

Reply:
I wondered if that was the case.  Playing devil's advocate, the type could change to an unsigned char, and pkcs11configMAX_SESSIONS could be set similarly...  I'd really like to see the code make the intention explicit.

Reply:
The type for CK_SESSION_HANDLE is defined in the PKCS #11 specification. I'm not sure I understand the second part, would you like to see a check that xSession is > 1?

Reply:
I guess I think the careful way to proceed is not to rely upon wrap-around, i.e., explicitly check before subtracting.  Sorry that my contrived scenario does not evoke the desired dread...

Reply:
I have updated this logic based on your feedback, now it uses a more explicit range of valid session handles